src/main/java/frc/robot/subsystems/color_wheel/commands/Spin.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.subsystems.color_wheel.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.color_wheel.ColorWheel;
+
+/**
+ * Spin the wheel 3 times.
+ */
+public class Spin extends CommandBase {
+
+    private final ColorWheel colorWheel;
+    private double power;
+    private String previousColor; //the first color the sensor sees.
+    private String currentColor; //the first color the sensor sees.
+    private boolean isPowerDecreased = false; //whether the power decrease has been executed already.
+    private int differentColorCounter = 0;
+
+    public Spin(ColorWheel colorWheel, double power) {
+        this.colorWheel = colorWheel;
+        this.power = power;
+        addRequirements(colorWheel);
+    }
+
+    @Override
+    public void initialize() {
+        colorWheel.updateSensor();
+        previousColor = colorWheel.getColorString();
+        currentColor = colorWheel.getColorString();
+        colorWheel.power(power);
+    }
+
+    @Override
+    public void execute() {
+        colorWheel.updateSensor();
+        currentColor = colorWheel.getColorString();
+        updateDifferentColorCount();
+        moderatePower();
+    }
+
+    @Override
+    public boolean isFinished() {
+        return differentColorCounter >= Constants.ColorWheel.REQUIRED_SPINS * Constants.ColorWheel.COLOR_WHEEL_SLOTS;
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        colorWheel.power(0);
+    }
+
+    public void updateDifferentColorCount() {
+        if (currentColor != previousColor) {
+            previousColor = currentColor;
+            differentColorCounter++;
+        }
+    }
+
+    public void moderatePower() {
+        if (differentColorCounter == Constants.ColorWheel.REQUIRED_SPINS * Constants.ColorWheel.COLOR_WHEEL_SLOTS - 1 && !isPowerDecreased) {
+            colorWheel.power(Constants.ColorWheel.REDUCE_POWER_BY * power);
+            isPowerDecreased = true;
+        }

```

Comment:
You know that `PercentOutput` won't continue to run unless you tell him (i.e you don't need the `isPowerDecreased` boolean, and you need to add else clause to apply the "not-decreased" power)